src/main/java/ch/njol/skript/SkriptCommand.java

```diff
@@ -497,20 +587,128 @@ private static List<File> getSubFiles(File file) {
 		return files;
 	}
 
-	private static @Nullable File getScriptFromArgs(CommandSender sender, String[] args) {
-		return getScriptFromArgs(sender, args, Skript.getInstance().getScriptsFolder());
+	private static File @Nullable [] getScriptsFromArgs(CommandSender sender, String[] args) {
+		return getScriptsFromArgs(sender, args, Skript.getInstance().getScriptsFolder());
+	}
+
+	private static File @Nullable [] getScriptsFromArgs(CommandSender sender, String[] args, File directoryFile) {
+		boolean enable = args[0].equals("enable");
+		Map<String, File> scripts = new HashMap<>();
+		for (String currentScript : parseAsCommaSeparatedList(false, true, args)) {
+			File thisScript = getScriptFromArg(sender, currentScript, directoryFile);
+			if (thisScript == null) {
+				// Always allow '/' and '\' regardless of OS
+				boolean directory = currentScript.endsWith("/") || currentScript.endsWith("\\") || currentScript.endsWith(File.separator);
+				Skript.error(sender, (directory ? m_invalid_folder : m_invalid_script).toString(currentScript));
+				continue;
+			}
+			scripts.put(currentScript, thisScript);
+		}
+
+		Map<String, File> filteredScripts = new HashMap<>();
+		Set<Entry<String, File>> entries = scripts.entrySet();
+		// Now we check to see if any of the files found is already set to be loaded if the directory was also provided
+		for (Entry<String, File> entry : entries) {
+			boolean add = true;
+			for (Entry<String, File> other : entries) {
+				if (entry.getKey().equals(other.getKey()))
+					continue;
+
+				if (entry.getKey().contains(other.getKey()) && entry.getValue().getPath().contains(other.getValue().getPath())) {
+					add = false;
+					break;
+				}
+			}
+			if (add && !filteredScripts.containsKey(entry.getKey())) {
+				boolean isDisabled = isScriptDisabled(entry.getKey());
+				// If we're enabling and this script is disabled, we can add it
+				// If we're disabling and this script is enabled, we can add it
+				if (enable == isDisabled)
+					filteredScripts.put(entry.getKey(), entry.getValue());
+			}
+		}
+		return filteredScripts.values().toArray(File[]::new);
+	}
+
+	/**
+	 * Checks if the provided {@link File} is disabled by checking if the file and the directory parents
+	 * are disabled via the prefix '-'
+	 * @param file
+	 * @return {@code true} if disabled
+	 */
+	public static boolean isScriptDisabled(File file) {
+		return isScriptDisabled(file.getPath());
+	}
+
+	/**
+	 * Checks if the provided {@code path} from a file is disabled by checking if the file and the directory parents
+	 * are disabled via the prefix '-'
+	 * @param path
+	 * @return {@code true} if disabled
+	 */
+	public static boolean isScriptDisabled(String path) {
+		String[] split = path.split("\\\\");
+		for (String segment : split) {
+			if (segment.startsWith("-"))
+				return true;
+		}
+		return false;
+	}
+
+	public static List<String> parseAsCommaSeparatedList(boolean reverse, boolean excludeEmptyArgs, String ... strings) {
```

Comment:
docs needed, this goes for all the other helpers too.
Comments within the code explaining what parts do would also be welcome.